### PR TITLE
Update ShareDB types

### PR DESCRIPTION
**Pull request by alecgibson**
Please see individual commit messages

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: (see commit messages)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/sharedb/lib/sharedb.d.ts [hunk 1]
```diff
@@ -133,6 +152,33 @@ export class Query extends EventEmitter {
     destroy(): void;
 }
 
+export type ReceivePresence<T> = (id: string, value: T) => void;
```
Comment (ericyhwang) [inline types/sharedb/lib/sharedb.d.ts]: I wasn't sure what this was at first, before noticing it's a function. A name like `ReceivePresenceListener` (or `ReceivePresenceCallback` like you suggested verbally) could make this clearer

**Review by ericyhwang — APPROVED** [review]
LGTM!

Note on the Authors update - Alec and Christina are both maintainers on `sharedb` itself, and they both know TS. So it makes sense to have them as authors on these type definitions as well.

The three of us discussed these type updates at our weekly PR review meeting, and Alec has been using these type additions internally for a while now.

**Review by ericyhwang — APPROVED** [review]
Re-LGTM

**Comment by alecgibson** [comment]
Ready to merge

**Pull request merged**